ui/appui-react/src/appui-react/framework/FrameworkDialogs.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { DialogChangedEvent, DialogInfo, DialogManagerBase } from "../dialog/DialogManagerBase";
+
+/** Modal Dialog Changed Event class.
+ * @public
+ */
+export class ModalDialogChangedEvent extends DialogChangedEvent { }
+
+/** Modeless Dialog Changed Event class.
+ * @public
+ */
+export class ModelessDialogChangedEvent extends DialogChangedEvent { }
+
+/** @public */
+export interface ModelessDialogInfo {
+  reactNode: React.ReactNode;
+  zIndex: number;
+  parentDocument: Document;
+}
+
+/**
+ * Manages dialog access
+ * @beta
+ */
+export interface FrameworkDialog {
+  /** @internal */
+  readonly dialogManager: DialogManagerBase;
+
+  /** Get the array of modal dialogs */
+  readonly dialogs: DialogInfo[];
+
+  /** Open a modal dialog
+   * @param dialog The Dialog to open
+   * @param id The id of the Dialog. If one is not provided, an id is generated.
+   * @param parentDocument The Document used to determine the owning window.
+   */
+  openDialog(dialog: React.ReactNode, id?: string, parentDocument?: Document): void;
```

Comment:
`UiFramework.dialogs.modal.openDialog()` is not too consistent with i.e. `UiFramework.childWindows.open()`
Would this be better?
```suggestion
  open(dialog: React.ReactNode, id?: string, parentDocument?: Document): void;
```

Reply:
Did many renames for consistency.